habitat/tasks/rearrange/marker_info.py

```diff
@@ -27,7 +27,9 @@ def __init__(self, offset_position, link_node, ao_parent, link_id):
         self.update()
 
     def set_targ_js(self, js):
-        self.ao_parent.joint_positions[self.joint_idx] = js
+        js_arr = self.ao_parent.joint_positions[:]
+        js_arr[self.joint_idx] = js
+        self.ao_parent.joint_positions = js_arr
```

Comment:
Why the extra logic?
If the previous code had an issue, I think you should write a unit test to make sure the same bug does not happen again

Reply:
This was due to some strange bug where the joint position was sometimes not set correctly. I will be challenging to setup an explicit test for this random situation.